src/background/tab-state.js

```diff
@@ -183,14 +184,22 @@ export default function TabState(initialState, onchange) {
       pendingAnnotationCountRequests.set(tabId, {
         cancel: () => {
           clearTimeout(timerId);
-          resolve(0);
+          reject(new BadgeRequestError('Badge request cancelled'));
```

Comment:
It seems like we use a mix of spellings of "cancelled" (one vs two Ls). I would suggest we favor the single-L version as various sources say that is more common in American English and is also what various web specifications use.

Reply:
OK.